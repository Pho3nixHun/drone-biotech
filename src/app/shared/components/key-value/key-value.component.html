<div
    class="duration-500 h-full flex flex-col justify-center"
    *transloco="let t"
>
    @if (label(); as label) {
        <p
            class="xl:hidden display uppercase text-gray-400 text-xs tracking-widest"
        >
            {{ t(label) }}
        </p>
    }
    <ng-content />
</div>
