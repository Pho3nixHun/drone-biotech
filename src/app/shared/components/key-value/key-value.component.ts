import { Component, input } from '@angular/core';
import { TranslocoModule } from '@jsverse/transloco';

/**
 * KeyValueComponent
 *
 * Type: Presentational
 *
 * Scope:
 * - Responsible for rendering an individual KeyValue pair.
 * - Displays the label and a paragraph via content projection.
 * - Manages the styling and the layout of a KeyValue pair.
 *
 * Out-of-Scope:
 * - Should not handle data fetching.
 * - Contains no business logic related to the testimonial.
 * - Does not manage or transform testimonial data.
 *
 * Purpose (optional):
 * - To render an individual Key-Value par in a predefined layout.
 */

@Component({
    selector: 'app-key-value',
    imports: [TranslocoModule],
    templateUrl: './key-value.component.html'
})
export class KeyValueComponent {
    public label = input.required<string>();
}
