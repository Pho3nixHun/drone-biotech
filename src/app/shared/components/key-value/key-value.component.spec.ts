import { ComponentFixture, TestBed } from '@angular/core/testing';

import { KeyValueComponent } from './key-value.component';
import { Component, input } from '@angular/core';
import { getTranslocoModule } from 'transloco-testing.module';

const en = {
    label: 'value',
};

@Component({
    template: `
        <app-key-value [label]="vm()">
            <p>Should be projected</p>
        </app-key-value>
    `,
})
class TestHostComponent {
    public vm = input.required<string>();
}

describe('KeyValueComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [
                KeyValueComponent,
                getTranslocoModule({
                    langs: { en: en },
                    translocoConfig: {
                        availableLangs: ['en'],
                        defaultLang: 'en',
                    },
                }),
            ],
            declarations: [TestHostComponent],
        }).compileComponents();
        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    it('should render the template when the VM is provided', () => {
        //Arrange
        fixture.componentRef.setInput('vm', en.label);

        //Act
        fixture.detectChanges();

        //Assert
        expect(compiled).toMatchSnapshot();
    });
});
