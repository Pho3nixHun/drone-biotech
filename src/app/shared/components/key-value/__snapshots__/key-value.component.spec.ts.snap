// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyValueComponent should render the template when the VM is provided 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-key-value
    ng-reflect-label="value"
  >
    <div
      class="duration-500 h-full flex flex-col justify-center"
    >
      <p
        class="xl:hidden display uppercase text-gray-400 text-xs tracking-widest"
      >
         value 
      </p>
      <!--container-->
      <p>
        Should be projected
      </p>
    </div>
    <!--container-->
  </app-key-value>
</div>
`;
